internal/service/appfabric/app_bundle_test.go

```diff
@@ -0,0 +1,194 @@
+			if err != nil {
+				return create.Error(names.AppFabric, create.ErrActionCheckingDestroyed, tfappfabric.ResNameAppBundle, rs.Primary.ID, err)
+			}
+
+			return create.Error(names.AppFabric, create.ErrActionCheckingDestroyed, tfappfabric.ResNameAppBundle, rs.Primary.ID, errors.New("not destroyed"))
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAppBundleExists(ctx context.Context, name string, appbundle *appfabric.GetAppBundleOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return create.Error(names.AppFabric, create.ErrActionCheckingExistence, tfappfabric.ResNameAppBundle, name, errors.New("not found"))
+		}
+		if rs.Primary.ID == "" {
+			return create.Error(names.AppFabric, create.ErrActionCheckingExistence, tfappfabric.ResNameAppBundle, name, errors.New("not set"))
+		}
+		conn := acctest.Provider.Meta().(*conns.AWSClient).AppFabricClient(ctx)
+		resp, err := conn.GetAppBundle(ctx, &appfabric.GetAppBundleInput{
+			AppBundleIdentifier: aws.String(rs.Primary.Attributes["arn"]),
+		})
+		if err != nil {
+			return create.Error(names.AppFabric, create.ErrActionCheckingExistence, tfappfabric.ResNameAppBundle, rs.Primary.ID, err)
+		}
+		*appbundle = *resp
+		return nil
+	}
+}
+
+func testAccPreCheck(ctx context.Context, t *testing.T) {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).AppFabricClient(ctx)
+	input := &appfabric.ListAppBundlesInput{}
+	_, err := conn.ListAppBundles(ctx, input)
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+func testAccCheckAppBundleNotRecreated(before, after *appfabric.GetAppBundleOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if before, after := aws.ToString(before.AppBundle.Arn), aws.ToString(after.AppBundle.Arn); before != after {
+			return create.Error(names.AppFabric, create.ErrActionCheckingNotRecreated, tfappfabric.ResNameAppBundle, before, errors.New("recreated"))
+		}
+		return nil
+	}
+}
+
+func testAccAppBundleImportStateIDFunc(ctx context.Context, resourceName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return "", fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return "", errors.New("No AppBundle ID set")
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).AppFabricClient(ctx)
+		appBundleARN := rs.Primary.Attributes["arn"]
+
+		_, err := conn.GetAppBundle(ctx, &appfabric.GetAppBundleInput{
+			AppBundleIdentifier: aws.String(appBundleARN),
+		})
+
+		if err != nil {
+			return "", err
+		}
+
+		return appBundleARN, nil
+	}
+}
+
+func testAccAppBundleConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_appfabric_app_bundle" "test" {
+	tags = {
+		Name = "AppFabricTesting"
+	}
+}
+`)
+}
```

Comment:
Add one more test called _full and ensure you can pass customer_managed_key_identifier and also ensure all fields are returned in the output. Idea is _basic test is too check minimum and _full test should check all fields possible (even optional fields)

Reply:
There are no required fields, so I just had :
testAccCheckAppBundleExists(ctx, resourceName, &appbundle),

Reply:
Could _basic be deleted altogether?